src/juju/reducer.js

```diff
@@ -86,6 +86,9 @@ export default function jujuReducer(state = defaultState, action) {
         break;
       case actionsList.populateMissingAllWatcherData:
         if (!draftState.modelWatcherData?.[payload.uuid]?.model) {
+          if (!draftState.modelWatcherData?.[payload.uuid]) {
+            draftState.modelWatcherData[payload.uuid] = {};
+          }
           draftState.modelWatcherData[payload.uuid].model = {};
         }
```

Comment:
This combination of negative logic and optional chaining is a bit hard to follow. I wonder if this could be rewritten in the positive, or some descriptive intermediary variables could be used like:
``` typescript
const existingUUID = draftState.modelWatcherData?.[payload.uuid];
const existingModel = existingUUID?.model;

if (!existingModel) {
  if (!existingUUID) {
    draftState.modelWatcherData[payload.uuid] = {};
  }
  draftState.modelWatcherData[payload.uuid].model = {};
}
```

Reply:
Good call, in the follow-up I've got some changes here so I'll apply this there to avoid conflicts.